server/src/main/webapp/WEB-INF/rails/webpack/views/components/buttons/index.tsx

```diff
@@ -108,10 +121,11 @@ abstract class Button extends RestyleViewComponent<Styles, Attrs> {
       <button {...Button.onlyHtmlAttrs(vnode.attrs)}
               {...dataTestIdAttrs(vnode.attrs)}
               onclick={clickHandler}
+              disabled={vnode.attrs.disabled || this.ajaxOperationMonitor() === OperationState.IN_PROGRESS}
               class={classnames(
                 this.css.button,
                 {[this.css.btnDropdown]: isDropdown},
-                Button.iconClass(this.ajaxInProgress, vnode.attrs.icon, this.css),
+                Button.iconClass(this.forceSpinner, vnode.attrs.icon, this.css),
```

Comment:
was there any specific reason to add another state `forceSpinner`? This can be achieved by `this.ajaxOperationMonitor() === OperationState.IN_PROGRESS` right? We can move this to a method and call the same at both places(line 124 and 128)?

Reply:
I did this to handle cases where there are two or more ajax operations possible. E.g. On the new Pipeline creation page, there are two possible ajax operations - `Save + Edit full config` or `Save + Run this pipeline`. If we bind to the operationMonitor, both the buttons will show the spinner. Since the `forceSpinner` is local to the button, it will show the spinner only on the button which was clicked.